src/components/PostsList.tsx

```diff
@@ -1,54 +1,54 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
 
 import classNames from 'classnames';
-import React from 'react';
-import { Post } from '../types/Post';
+import { FC } from 'react';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { setSelectedPost } from '../features/selectedPost';
 
-type Props = {
-  posts: Post[];
-  selectedPostId?: number;
-  onPostSelected: (post: Post | null) => void;
-};
+export const PostsList: FC = () => {
+  const posts = useAppSelector(state => state.posts.posts);
```

Comment:
It's a good practice to separate the logic of fetching data from the component. Consider creating a custom hook for fetching posts. This way, the PostsList component will not be concerned with how the data is fetched, making the code cleaner and easier to test.